src/libcore/hint.rs

```diff
@@ -21,9 +21,8 @@ use intrinsics;
 /// difficult-to-debug problems.
 ///
 /// Use this function only when you can prove that the code will never call it.
-///
-/// The [`unreachable!()`] macro is the safe counterpart of this function, which
-/// will panic instead when executed.
+/// Otherwise, consider using the [`unreachable!()`] macro, which does not allow
```

Comment:
This is good, but can you remove the `()s`, both here and in the link target? I know they were there previously, but the style is not to include them when referring to the names.

Reply:
I added another commit to address this.